rpc/core/events.go

```diff
@@ -106,9 +105,7 @@ func Subscribe(ctx *rpctypes.Context, query string) (*ctypes.ResultSubscribe, er
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to parse query")
 	}
-	subscribeTimeout := rpcserver.WriteTimeout() / 2
-	subCtx, cancel := context.WithTimeout(context.Background(), subscribeTimeout)
-	defer cancel()
+	subCtx := context.TODO()
```

Comment:
TODO: get ctx from ctx *rpctypes.Context

Reply:
So the subscribeTimeout isn't absolutely necessary here? That simplifies this a lot ...

Reply:
No, looks like we need this timeout if we want to properly respond to the client's request.

Reply:
https://github.com/tendermint/tendermint/pull/3443/commits/db0986e8ebc7fd4f1190d083227d11d9df97f011